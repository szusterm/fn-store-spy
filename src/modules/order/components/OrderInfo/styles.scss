@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins/breakpoints';

@import '~scss/_shadow';
@import '~scss/_transition';
@import '~scss/_opacity-colors';
@import '~scss/_center-items';

.order-info {
	$list-width: 400px;

	position: fixed;
	top: 0;
	bottom: 0;
	left: auto;
	right: 0;
	z-index: 20;

	height: 100%;

	@include white-background(1);
	@include shadow(md);

	transition-property: transform;
	@include transition(md);

	@include media-breakpoint-down(sm) {width: 100%;}
	@include media-breakpoint-between(sm, md) {width: 70%;}
	@include media-breakpoint-between(md, lg) {width: 40%;}
	@include media-breakpoint-up(lg) {width: 30%;}

	&--hidden {
		transform: translate(100%);
	}

	&--components {
		height: 100%;

		display: flex;
		flex-direction: column;

		.components {
			&--button-box {
				@include center-items();
				padding-top: 5px;
			}

			&--list {
				height: 100%;
			}

			&--confirmation-box {
				@include center-items();
				justify-content: space-around;
				padding-bottom: 60px;
			}
		}
	}
}