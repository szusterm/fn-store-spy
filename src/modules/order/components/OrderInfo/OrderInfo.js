import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {closeOrderList} from '../../actions';

import BlackCurtain from '../../../../components/BlackCurtain';
import HideOrderButton from '../HideOrderButton';
import ItemOrderCard from '../ItemOrderCard';

import './styles.scss';

export class OrderInfo extends Component {

	checkThatOrderIsEmpty() {
		const {items} = this.props;

		const isEmpty = (items.length === 0);

		if (isEmpty) {
			this.props.closeOrderList();
		}

		return isEmpty;
	}

	getItemColsWidth() {
		const {items} = this.props;
		return (items.length <= 9) ? 4 : 3;
	}

	render() {
		const {items, listOpened, closeOrderList} = this.props;

		return (
			<div>
				<BlackCurtain
					show={listOpened}
					onClick={closeOrderList}
				/>
				<div className={`order-info ${(!listOpened || this.checkThatOrderIsEmpty()) && 'order-info--hidden'}`}>
					<div className={'container-fluid'}>
						<div className={'row order-info--button-box'}>
							<div className={'col-8'}>
								<HideOrderButton/>
							</div>
						</div>
						<div className={'row order-info--list'}>
							{
								items.map((item, index) => {
									const {_id, imageSrc, rarity} = item;

									return (
										<div
											key={_id}
											className={`col-${this.getItemColsWidth()} list--item-box`}
										>
											<ItemOrderCard
												index={index}
												imageSrc={imageSrc}
												rarity={rarity}
											/>
										</div>
									);
								})
							}
						</div>
					</div>
					<div className={'order-info--confirmation-box'}/>
				</div>
			</div>
		);
	}
}

OrderInfo.propTypes = {
	items: PropTypes.array, //redux
	listOpened: PropTypes.bool, //redux
	closeOrderList: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {items, listOpened} = state.order;

	return {items, listOpened};
};

const mapDispatchToProps = {closeOrderList};

export default connect(mapStateToProps, mapDispatchToProps)(OrderInfo);