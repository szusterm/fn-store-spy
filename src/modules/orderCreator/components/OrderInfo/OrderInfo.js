import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {closeOrderList} from '../../actions';

import OrderList from '../OrderList';
import BlackCurtain from '../../../../components/BlackCurtain';
import HideOrderButton from '../HideOrderButton';
import FullOrderCount from '../FullOrderCount';
import SendOrderButton from '../../../orderConfirming/components/SendOrderButton';

import './styles.scss';

export class OrderInfo extends Component {
	isOrderEmpty() {
		const {items} = this.props;

		const isEmpty = (items.length === 0);

		if (isEmpty) {
			this.props.closeOrderList();
		}

		return isEmpty;
	}

	render() {
		const {listOpened, closeOrderList} = this.props;

		return (
			<div>
				<BlackCurtain
					show={listOpened}
					onClick={closeOrderList}
				/>
				<div className={`order-info ${(!listOpened || this.isOrderEmpty()) && 'order-info--hidden'}`}>
					<div className={'container-fluid order-info--components'}>
						<div className={'row components--button-box'}>
							<div className={'col-8'}>
								<HideOrderButton/>
							</div>
						</div>
						<div className={'row components--list'}>
							<div className={'col-12'}>
								<OrderList/>
							</div>
						</div>
						<div className={'row components--confirmation-box'}>
							<div className={'col-12'}>
								<FullOrderCount/>
							</div>
							<div className={'col-9'}>
								<SendOrderButton/>
							</div>
						</div>
					</div>
				</div>
			</div>
		);
	}
}

OrderInfo.propTypes = {
	items: PropTypes.array, //redux
	listOpened: PropTypes.bool, //redux
	closeOrderList: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {items, listOpened} = state.order;
	return {items, listOpened};
};

const mapDispatchToProps = {closeOrderList};

export default connect(mapStateToProps, mapDispatchToProps)(OrderInfo);