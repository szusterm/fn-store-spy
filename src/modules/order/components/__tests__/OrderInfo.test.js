import React from 'react';
import {shallow} from 'enzyme';

import {OrderInfo} from '../OrderInfo/OrderInfo';

const setup = (propOverrides) => {
	const props = Object.assign({

	}, propOverrides);

	const wrapper = shallow(<OrderInfo {...props}/>);

	return {props, wrapper};
};

describe('Order Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});
});