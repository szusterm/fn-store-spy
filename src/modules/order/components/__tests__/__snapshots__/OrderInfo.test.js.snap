// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderInfo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BlackCurtain />,
        <div
          className="order-info order-info--hidden"
        >
          <div
            className="container-fluid"
          >
            <div
              className="row order-info--button-box"
            >
              <div
                className="col-8"
              >
                <Connect(HideOrderButton) />
              </div>
            </div>
            <div
              className="row order-info--list"
            >
              <div
                className="col-12"
              >
                <Connect(OrderList) />
              </div>
            </div>
          </div>
          <div
            className="order-info--confirmation-box"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClick": undefined,
          "show": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="container-fluid"
            >
              <div
                className="row order-info--button-box"
              >
                <div
                  className="col-8"
                >
                  <Connect(HideOrderButton) />
                </div>
              </div>
              <div
                className="row order-info--list"
              >
                <div
                  className="col-12"
                >
                  <Connect(OrderList) />
                </div>
              </div>
            </div>,
            <div
              className="order-info--confirmation-box"
            />,
          ],
          "className": "order-info order-info--hidden",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="row order-info--button-box"
                >
                  <div
                    className="col-8"
                  >
                    <Connect(HideOrderButton) />
                  </div>
                </div>,
                <div
                  className="row order-info--list"
                >
                  <div
                    className="col-12"
                  >
                    <Connect(OrderList) />
                  </div>
                </div>,
              ],
              "className": "container-fluid",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="col-8"
                  >
                    <Connect(HideOrderButton) />
                  </div>,
                  "className": "row order-info--button-box",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Connect(HideOrderButton) />,
                    "className": "col-8",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="col-12"
                  >
                    <Connect(OrderList) />
                  </div>,
                  "className": "row order-info--list",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Connect(OrderList) />,
                    "className": "col-12",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "order-info--confirmation-box",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BlackCurtain />,
          <div
            className="order-info order-info--hidden"
          >
            <div
              className="container-fluid"
            >
              <div
                className="row order-info--button-box"
              >
                <div
                  className="col-8"
                >
                  <Connect(HideOrderButton) />
                </div>
              </div>
              <div
                className="row order-info--list"
              >
                <div
                  className="col-12"
                >
                  <Connect(OrderList) />
                </div>
              </div>
            </div>
            <div
              className="order-info--confirmation-box"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClick": undefined,
            "show": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="container-fluid"
              >
                <div
                  className="row order-info--button-box"
                >
                  <div
                    className="col-8"
                  >
                    <Connect(HideOrderButton) />
                  </div>
                </div>
                <div
                  className="row order-info--list"
                >
                  <div
                    className="col-12"
                  >
                    <Connect(OrderList) />
                  </div>
                </div>
              </div>,
              <div
                className="order-info--confirmation-box"
              />,
            ],
            "className": "order-info order-info--hidden",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="row order-info--button-box"
                  >
                    <div
                      className="col-8"
                    >
                      <Connect(HideOrderButton) />
                    </div>
                  </div>,
                  <div
                    className="row order-info--list"
                  >
                    <div
                      className="col-12"
                    >
                      <Connect(OrderList) />
                    </div>
                  </div>,
                ],
                "className": "container-fluid",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="col-8"
                    >
                      <Connect(HideOrderButton) />
                    </div>,
                    "className": "row order-info--button-box",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Connect(HideOrderButton) />,
                      "className": "col-8",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="col-12"
                    >
                      <Connect(OrderList) />
                    </div>,
                    "className": "row order-info--list",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Connect(OrderList) />,
                      "className": "col-12",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "order-info--confirmation-box",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
